src/gallium/drivers/lima/lima_draw.c

```diff
@@ -908,6 +920,11 @@ lima_draw_vbo(struct pipe_context *pctx, const struct pipe_draw_info *info)
       ctx->const_buffer[PIPE_SHADER_FRAGMENT].dirty = false;
    }
 
+   if (!ctx->num_draws ||
+      (ctx->dirty & LIMA_CONTEXT_DIRTY_TEXTURES)) {
+      lima_update_textures(ctx);
```

Comment:
You need a lima_ctx_buff_state here for the texture desc. For example, first draw need texture A, so store its desc in 0 offset, second draw need texture B, so it has to store its desc after the first desc, then a flush for a frame can use the right desc for each draw.

Reply:
OK, will do.

Reply:
Hm, I don't understand how it's supposed to work. Do I need to reserve space for descriptors for several draws in pp_buffer?

Reply:
Yes, you need to reserve that. See this commit for adding a lima_ctx_buff_state:
https://github.com/yuq/mesa-lima/commit/7dded4f568c199c5714c4cf0ad1c3c1851951c06

Reply:
Hm, so it'll fail if buffer overflows?

Reply:
Yes, but no check for now. I'm considering refine this.